test/e2e/features_test.go

```diff
@@ -703,3 +703,95 @@ func TestDefaultIngressClass(t *testing.T) {
 		return false
 	}, ingressWait, time.Second)
 }
+
+// TestMissingCRDsDontCrashTheController ensures that in case of missing CRDs installation in the cluster, specific
+// controllers are disabled, this fact is properly logged, and the controller doesn't crash.
+func TestMissingCRDsDontCrashTheController(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	clusterBuilder := kind.NewBuilder()
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.ParseTolerant(clusterVersionStr)
+		require.NoError(t, err)
+		clusterBuilder.WithClusterVersion(clusterVersion)
+	}
+	cluster, err := clusterBuilder.Build(ctx)
+	require.NoError(t, err)
+	var addons []clusters.Addon
+	addons = append(addons, metallb.New())
+
+	if b, err := loadimage.NewBuilder().WithImage(imageLoad); err == nil {
+		addons = append(addons, b.Build())
+	}
+
+	builder := environments.NewBuilder().WithExistingCluster(cluster).WithAddons(addons...)
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+
+	t.Logf("building cleaner to dump diagnostics...")
+	cleaner := clusters.NewCleaner(env.Cluster())
+	defer func() {
+		if t.Failed() {
+			output, err := cleaner.DumpDiagnostics(ctx, t.Name())
+			t.Logf("%s failed, dumped diagnostics to %s", t.Name(), output)
+			assert.NoError(t, err)
+		}
+		assert.NoError(t, cleaner.Cleanup(ctx))
+		assert.NoError(t, cluster.Cleanup(ctx))
```

Comment:
Do we really need to use the cleaner if we tear down the cluster? I don't see a point in this **unless** we introduce the concept of reusing the cluster (but as far as I can see none of the e2e tests here use it so it would be a significant patch on its own).

Hence my suggestion to just use the cluster cleaner (and perhaps) we'd need to consider doing this in other E2Es as well 🤔 

```suggestion
		assert.NoError(t, cluster.Cleanup(ctx))
```

Reply:
Hm, yup, I think that we might have this issue in every E2E since the other test cases use only cleaner. 🤔 Do you want me to change it in this PR or rather we should have a separate one (I'd vote for a separate one)?

Reply:
f3bc3b9

Reply:
Separate one would be better to address this